Source/ProgressionSystemRuntime/Public/Data/PSWorldSubsystem.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "PSTypes.h"
+#include "Subsystems/WorldSubsystem.h"
+#include "PSWorldSubsystem.generated.h"
+
+
+/**
+ * Implements the world subsystem to access different components in the module 
+ */
+UCLASS(BlueprintType, Blueprintable, Config = "ProgressionSystem", DefaultConfig)
+class PROGRESSIONSYSTEMRUNTIME_API UPSWorldSubsystem : public UWorldSubsystem
+{
+	GENERATED_BODY()
+
+public:
+	DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FCurrentRowDataChanged, const FPlayerTag, SavedProgressionRowData);
+
+	/** Returns this Subsystem, is checked and wil crash if can't be obtained.*/
+	static UPSWorldSubsystem& Get();
+	static UPSWorldSubsystem& Get(const UObject& WorldContextObject);
+
+	const FPSRowData& GetCurrentRow() const;
+
+	/* Delegate for informing row data changed */
+	UPROPERTY(BlueprintAssignable, Category = "C++")
+	FCurrentRowDataChanged OnCurrentRowDataChanged;
+
+	/** Returns the data asset that contains all the assets of Progression System game feature.
+	 * @see UPSWorldSubsystem::PSDataAssetInternal. */
+	UFUNCTION(BlueprintCallable, BlueprintPure, Category = "C++")
+	const class UPSDataAsset* GetPSDataAsset() const;
+
+	/** Returns a progression System component reference */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	FORCEINLINE class UPSHUDComponent* GetProgressionSystemHUDComponent() const { return PSHUDComponentInternal; }
+
+	/** Returns a current save game instance */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	FORCEINLINE UPSSaveGameData* GetCurrentSaveGameData() const { return SaveGameInstanceInternal; }
```

Comment:
Please add `class` 'forward', otherwise it will fail the cook
```
FORCEINLINE [class] UPSSaveGameData* ...
```

Reply:
added